mslib/mswms/utils.py

```diff
@@ -309,153 +542,12 @@ def get_style_parameters(dataname, style, cmin, cmax, data):
-             (0.82333333333333336, 0.40000000000000002, 1.0, 1.0),
-             (0.64666666666666672, 0.25, 1.0, 1.0),
-             (0.46999999999999997, 0.10000000000000001, 1.0, 1.0),
-             (0.69999999999999996, 1.0, 1.0, 1.0),
-             (0.0, 0.20000000000000001, 1.0, 1.0),
-             (0.65000000000000002, 1.0, 0.65000000000000002, 1.0),
-             (0.0, 0.69999999999999996, 0.0, 1.0),
-             (1.0, 1.0, 0.0, 1.0),
-             (1.0, 0.73529411764705888, 0.0, 1.0),
-             (1.0, 0.46323529411764708, 0.0, 1.0),
-             (1.0, 0.21568627450980393, 0.0, 1.0),
-             (1.0, 0.034313725490196068, 0.0, 1.0),
-             (0.8294117647058824, 0.0, 0.071078431372549017, 1.0),
-             (0.61176470588235299, 0.0, 0.16176470588235295, 1.0),
-             (0.40000000000000002, 0.0, 0.25, 1.0),
-             ], name="n2_map")
-        cmap.set_over((0.8, 0.8, 0.8, 1.0))
-        clev = np.arange(0., 8.5 / 1e4, 0.5 / 1e4)
-        norm = matplotlib.colors.BoundaryNorm(clev, cmap.N)
-    elif style == "tropopause_altitude":
-        cmap = matplotlib.pyplot.cm.terrain
-        norm = None
-        clev = np.arange(5, 18.1, 0.25)
-    elif style == "log_ice_cloud":
-        cmap = matplotlib.pyplot.cm.colors.ListedColormap(
-            [(0.8, 0.8, 0.8, 1.0),
-             (1., 1., 1.0, 1.0),
-             (0., 0., 1.0, 1.0),
-             (0., 0.949, 0.988, 1.0),
-             (0., 1., 0., 1.0),
-             (1., 1.0, 0.0, 1.0),
-             (1.0, 0., 0.0, 1.0),
-             (0.212, 0.114, 0.075, 1.0),
-             ], name="ice_map_log")
-        if dataname == "number_concentration_of_ice_crystals_in_air":
-            clev = np.append(np.array([-1e31, -0.1]),
-                             get_log_levels(1e-4, 100., 7))
-            ticks = np.append(np.array([0.]), get_log_levels(1e-4, 100., 7))
-        if dataname == "cloud_ice_mixing_ratio":
-            clev = np.append(np.array([-1e31, -0.1]),
-                             get_log_levels(1e-3, 1e3, 7))
-            ticks = np.append(np.array([0.]), get_log_levels(1e-3, 1e3, 7))
-        norm = matplotlib.colors.BoundaryNorm(clev, cmap.N)
-    elif style == "ice_cloud":
-        cmap = matplotlib.pyplot.cm.colors.ListedColormap(
-            [(1., 1., 1.0, 1.0),
-             (0., 0., 1.0, 1.0),
-             (0., 0.949, 0.988, 1.0),
-             (0., 1., 0., 1.0),
-             (1., 1.0, 0.0, 1.0),
-             (1.0, 0., 0.0, 1.0),
-             (0.212, 0.114, 0.075, 1.0),
-             ], name="ice_map")
-        if dataname == "mean_mass_radius_of_cloud_ice_crystals":
-            clev = np.array([0, 1, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100])
-        norm = matplotlib.colors.BoundaryNorm(clev, cmap.N)
-    else:
-        raise RuntimeError(f"Illegal plotting style?! ({style})")
+    try:
+        cmin, cmax, clev, cmap, norm, ticks = \
+            _styles[style](dataname, style, cmin, cmax, cmap, data)
```

Comment:
ok, here are always 6 arguments passed and there are few by a name convention which have 7, which get translated